### ISSUE TITLE
Vuex and composition API

### ISSUE THREAD

**Issue by mikimoresi** [shuidi-fed/vapper#46]
Due to the necessary createStore function, based on this possible solution https://github.com/vuejs/composition-api/issues/8#issuecomment-506239327 
it appear to be not possible using store properly with composition API.
Am I missing something?
Thank you

**Comment by HcySunYang** [comment]
If I didn't get it wrong, you just need:

```js
import { getCurrentInstance } from '@vue/composition-api'
export function useStore() {
  const vm = getCurrentInstance()
  return vm.$store
}
```

**Comment by mikimoresi** [comment]
Thank you Sun, I still need to get more used to node mechanisms,
following your advice now is working way better but I'm trying to centralize all computed store values in one object but it don't react.

if i use

`
setup(props,{root}) {
	var store =  useStore().state;
        const changing_view = computed(() => store.main_store.changing_view);
}
`
my component correctly reacts to 'changing_view' values

while if i try to do something like 

`
setup(props,{root}) {
	var store =  useStore().state;
        const changing_view = computed(() => store.main_store.changing_view);
        const state = {
              changing_view:changing_view
        }
}
`

the component is not reacting anymore to state.changing_view.

Do you have any hint on this?

Thank you again

**Comment by HcySunYang** [comment]
Sorry, I don’t know what the problem is, maybe you should go to https://github.com/vuejs/composition-api to submit an issue.

**Comment by dmitry** [comment]
@mikimoresi have you found the solution to your question?

**Comment by mikimoresi** [comment]
hello @dmitry,
yes, i got it.
Here a repository to show how to properly implement composition API
https://github.com/mikimoresi/vapcompvuex

The main difference sith the proposed solution is in the store, I have had to use 
```js
export const store = createStore()
export const useStore = () => store
```
instead of the getCurrentInstance() method.

Anyway lately I found there are several limitation in the composition API such as problem with external resources imported via webpack alias (https://github.com/vuejs/composition-api/issues/228) that forced me to abandon my proposal of using the composition API. I will switch to it when Vue 3 will be officially released and solid.
